app/javascript/widget/helpers/availabilityHelpers.js

```diff
@@ -0,0 +1,214 @@
+import { utcToZonedTime } from 'date-fns-tz';
+
+// Constants
+const DAYS_IN_WEEK = 7;
+const MINUTES_IN_HOUR = 60;
+const MINUTES_IN_DAY = 24 * 60;
+
+/**
+ * Get date in timezone
+ * @param {Date|string} time
+ * @param {string} utcOffset
+ * @returns {Date}
+ */
+const getDateInTimezone = (time, utcOffset) => {
```

Comment:
Have we handled the invalid utcOffset case? If the utcOffset is always valid then there's no issue. Perhaps we should add a validation here.

```
const getDateInTimezone = (time, utcOffset) => {
    try {
      const dateString = time instanceof Date ? time.toISOString() : time;
      return utcToZonedTime(dateString, utcOffset);
    } catch (error) {
      console.warn(`Invalid timezone: ${utcOffset}, falling back to UTC`);
      return utcToZonedTime(dateString, 'UTC');
    }
  };
  ```